components/encryption/src/file_dict_file.rs

```diff
@@ -172,21 +173,28 @@ impl FileDictionaryFile {
                 // parse the remained records
                 while !remained.is_empty() {
                     // If the parsing get errors, manual intervention is required to recover.
-                    let (used_size, file_name, mode) = Self::parse_next_record(remained)?;
-                    remained.consume(used_size);
-                    match mode {
-                        LogRecord::INSERT(info) => {
-                            file_dict.files.insert(file_name, info);
-                        }
-                        LogRecord::REMOVE => {
-                            let original = file_dict.files.remove(&file_name);
-                            if original.is_none() {
-                                return Err(box_err!(
-                                    "Try to recovery from log file but remove a null entry, file name: {}",
-                                    file_name
-                                ));
+                    match Self::parse_next_record(remained) {
+                        Ok((used_size, file_name, mode)) => {
+                            remained.consume(used_size);
```

Comment:
I might misread the code, but `parse_next_record` also consume the bytes. It looks like we double consume the size here?

Reply:
`parse_next_record` just borrow the `remained` content.